snmp/datadog_checks/snmp/config.py

```diff
@@ -151,10 +151,13 @@ def __init__(
         if not self.metrics and not profiles_by_oid and not profile:
             raise ConfigurationError('Instance should specify at least one metric or profiles should be defined')
 
-        self.scalar_oids, self.next_oids, self.bulk_oids, self.parsed_metrics = self.parse_metrics(self.metrics)
+        parsed_scalar_oids, parsed_next_oids, parsed_bulk_oids, self.parsed_metrics = self.parse_metrics(self.metrics)
         tag_oids, self.parsed_metric_tags = self.parse_metric_tags(metric_tags)
         if tag_oids:
-            self.scalar_oids.extend(tag_oids)
+            parsed_scalar_oids.extend(tag_oids)
+
+        self.oid_config = OIDConfig()
+        self.oid_config.add_parsed_oids(parsed_scalar_oids, parsed_next_oids, parsed_bulk_oids)
```

Comment:
With my suggestions below this would become...

```suggestion
        self.oid_config.add(scalar=scalar_oids, next=next_oids, bulk=bulk_oids)
```